go/host/enclave/guardian.go

```diff
@@ -439,7 +441,9 @@ func (g *Guardian) submitL1Block(block *common.L1Block, isLatest bool) (bool, er
 		}
 	}
 
-	resp, err := g.enclaveClient.SubmitL1Block(context.Background(), block.Header(), txWithReceipts, isLatest)
+	_, rollupTxs, blobsAndHashes, contractAddressTxs := g.sl.L1Publisher().ExtractRelevantTenTransactions(block)
+	resp, err := g.enclaveClient.SubmitL1Block(context.Background(), block.Header(), txWithReceipts, blobsAndHashes)
```

Comment:
I propose a change , which I think will clarify the flow.

The `TxAndReceipt` struct becomes `TxAndReceiptAndBlobs`
Basically, when we read the relevant txs from the block, we also get the blobs and add them to this data structure.
This will go into the enclave via `SubmitL1Block`, so there is no more necessity to pass the blobs separately, and you'll know exactly which tx they belong to. So it can simplify the logic there a bit as well